hera_sim/tests/test_foregrounds.py

```diff
@@ -1,220 +1,138 @@
+    assert vis.shape == (lsts.size, freqs.size)
 
-        # check the shape
-        self.assertEqual(vis.shape, (lsts.size, freqs.size))
-
-        # test that foregrounds show up at positive fringe-rates
-
-        # make a purely EW baseline
-        bl_vec = (100.0, 0.0)
-
-        # use a gaussian fringe filter with a width of 10 uHz
-        fringe_filter_kwargs = {"fringe_filter_type": "gauss", "fr_width": 1e-5}
 
+@pytest.mark.parametrize("model", ["pntsrc", "diffuse"])
+def test_foreground_autos_are_real(freqs, lsts, Tsky_mdl, omega_p, model):
+    bl_vec = [0, 0, 0]
+    if model == "pntsrc":
+        vis = foregrounds.pntsrc_foreground(lsts=lsts, freqs=freqs, bl_vec=bl_vec)
+    elif model == "diffuse":
         vis = foregrounds.diffuse_foreground(
-            lsts,
-            freqs,
-            bl_vec,
-            Tsky_mdl=Tsky_mdl,
-            omega_p=omega_p,
-            fringe_filter_kwargs=fringe_filter_kwargs,
+            lsts=lsts, freqs=freqs, bl_vec=bl_vec, Tsky_mdl=Tsky_mdl, omega_p=omega_p
         )
-
-        # transform the visibility to FR-freq space
-        dfft = FFT(vis * dspec.gen_window("blackmanharris", len(vis))[:, None], axis=0)
-
-        # calculate the fringe rates
-        frates = np.fft.fftshift(
-            np.fft.fftfreq(len(lsts), np.diff(lsts)[0] * u.sday.to("s") / (2 * np.pi))
+    assert np.allclose(vis.imag, 0)
+    # import uvtools, pylab as plt
+    # uvtools.plot.waterfall(vis, mode='log'); plt.colorbar(); plt.show()
+
+
+@pytest.mark.parametrize("orientation", ["east", "west", "north"])
+@pytest.mark.parametrize(
+    "model", ["pntsrc", "diffuse"],
+)
+def test_foreground_orientation(freqs, lsts, Tsky_mdl, omega_p, model, orientation):
+    baselines = {"east": [100, 0, 0], "west": [-100, 0, 0], "north": [0, 100, 0]}
+    bl_vec = baselines[orientation]
+    fringe_filter_kwargs = {"fringe_filter_type": "gauss", "fr_width": 1e-5}
+    kwargs = dict(freqs=freqs, lsts=lsts, bl_vec=bl_vec)
+    # XXX Can this be made into a function to avoid repeated code?
+    if model == "diffuse":
+        model = foregrounds.diffuse_foreground
+        kwargs.update(
+            dict(
+                Tsky_mdl=Tsky_mdl,
+                omega_p=omega_p,
+                fringe_filter_kwargs=fringe_filter_kwargs,
+            )
         )
-
-        max_frate = frates[np.argmax(np.abs(dfft[:, 0]))]
-        nt.assert_true(max_frate > 0)
-
-        # now test that they show up at negative fringe-rates for
-        # an oppositely oriented baseline
-        bl_vec = (-100.0, 0.0)
-
-        vis = foregrounds.diffuse_foreground(
-            lsts,
-            freqs,
-            bl_vec,
-            Tsky_mdl=Tsky_mdl,
-            omega_p=omega_p,
-            fringe_filter_kwargs=fringe_filter_kwargs,
+    elif model == "pntsrc":
+        # XXX point source foregrounds do not behave correctly in fringe-rate
+        # Is there a way to parametrize this that gives an xfail when point source
```

Comment:
yes, see https://docs.pytest.org/en/stable/example/parametrize.html#parametrizing-conditional-raising

Reply:
argh, so this opened up a small can of worms. it looks like the point-source foregrounds get the fringe-rate behavior wrong for every baseline except the 100-m eastward baseline. so this isn't consistently "failing" as I had expected it to. I've made a little hack to deal with this, but I don't really like it.